runtime/vm/stringhelpers.cpp

```diff
@@ -414,38 +414,46 @@ getStringUTF8Length(J9VMThread *vmThread, j9object_t string)
 UDATA 
 verifyQualifiedName(J9VMThread *vmThread, j9object_t string)
 {
+	J9JavaVM* vm = vmThread->javaVM;
 	UDATA unicodeLength = J9VMJAVALANGSTRING_LENGTH(vmThread, string);
 	j9object_t unicodeBytes = J9VMJAVALANGSTRING_VALUE(vmThread, string);
-	UDATA i = 0;
+	char utf8NameStackBuffer[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
+	J9CfrConstantPoolInfo classNameInfo = {0, 0, 0, 0, 0, NULL, 0};
 	UDATA rc = CLASSNAME_VALID_NON_ARRARY;
+	IDATA arity = 0;
+
+	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	if (unicodeLength == 0) {
+	if (0 == unicodeLength) {
 		return CLASSNAME_INVALID;
 	}
 
 	if (IS_STRING_COMPRESSED(vmThread, string)) {
 		if ('[' == J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, 0)) {
```

Comment:
The array `[` prefix is handled by `checkNameImpl`.  No need to handle it here as well

Reply:
Agree to remove the check of `[` but we need to map `arity `(return value from checkNameImpl) to the correct return code `rc` as follows:
```
1) if arity = -1, then rc = CLASSNAME_INVALID (0) = arity + 1
2) if arity = 0, then rc = CLASSNAME_VALID_NON_ARRARY (1) = arity + 1
3) if arity > 0, then rc = arity + 1 (e.g. CLASSNAME_VALID_ARRARY)
```
Will double-check that.

Reply:
Already removed the check of `[`.